src/map_cell.cpp

```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+#include <array>
+#include <memory>
+#include <simulator/map_cell.hpp>
+#include <simulator/bug.hpp>
+
+namespace simulation
+{
+
+MapCell::MapCell(const std::shared_ptr<Bug> bug, const AdyacentCellsReference& c_ref) 
+  : _bug(bug), _adyacent_cells{c_ref.up, c_ref.left, c_ref.down, c_ref.right} {}
+
+MapCell::MapCell(const std::shared_ptr<Bug> bug) 
+  : _bug(bug), _adyacent_cells{nullptr, nullptr, nullptr, nullptr} {}
```

Comment:
Maybe a constructor for AdyacentCellsReference that creates a null object for this case?

Reply:
Same as above, but now that I'm looking at this initialization more carefully, explicitly initializing `_adyacents_cells` is not needed, as shared pointers in the array will be set to `nullptr` anyway.

I've tried this snippet to make sure (I should've written a test instead, I know).
```
#include <iostream>
#include <array>
#include <memory>

struct CustomObject {};

class TestClass {
 public:
  std::array<std::shared_ptr<CustomObject>, 4> arr;
};

int main() {
  TestClass my_class{};
  for(auto el: my_class.arr) {
    if(el == nullptr) {
      printf("Nullptr\n");
    }
  }
  return 0;
}
```